ta/remoteproc/remoteproc_core.c

```diff
@@ -0,0 +1,1212 @@
+		return res;
+	}
+
+	res = get_hash_table(ctx);
+	if (res)
+		return res;
+
+	/* Get the number of firmwares to load */
+
+	res = remoteproc_get_tlv(ctx->tlr, RPROC_TLV_NUM_IMG,
+				 &tlv, &length);
+	if (res)
+		goto out;
+	if (length != sizeof(uint8_t)) {
+		res = TEE_ERROR_BAD_FORMAT;
+		goto out;
+	}
+
+	num_img = *tlv;
+	ctx->rsc_pa = 0;
+	ctx->rsc_size = 0;
+
+	for (i = 0; i < num_img; i++) {
+		res = get_tlv_images_type(ctx, num_img, i, &img_type);
+		if (res)
+			goto out;
+		if (img_type != REMOTEPROC_ELF_TYPE) {
+			res = TEE_ERROR_BAD_FORMAT;
+			goto out;
+		}
+
+		res = get_tlv_images_size(ctx, num_img, i, &img_size);
+		if (res)
+			goto out;
+
+		res = e32_parser_load_elf_image(ctx->fw_img + offset, img_size,
+						remoteproc_load_segment, ctx);
+		if (res)
+			goto out;
+
+		/* Take opportunity to get the resource table address */
+		res = remoteproc_parse_rsc_table(ctx, ctx->fw_img + offset,
+						 img_size, &rsc_pa, &rsc_size);
+		if (res != TEE_SUCCESS && res != TEE_ERROR_NO_DATA)
+			goto out;
+
+		if (res != TEE_ERROR_NO_DATA) {
+			/*
+			 * Only one resource table is supported, check that no
+			 * other one has been declared in a previous loaded
+			 * firmware.
+			 */
+			if (ctx->rsc_pa || ctx->rsc_size) {
+				EMSG("More than one resource table found");
+				res = TEE_ERROR_BAD_FORMAT;
+				goto out;
+			}
+			ctx->rsc_pa = rsc_pa;
+			ctx->rsc_size = rsc_size;
+		}
+
+		offset += img_size;
+	}
+
+	/*
+	 * If res == TEE_ERROR_NO_DATA then no resource table found.
+	 * Force to TEE_SUCCESS as the resource table is optional.
+	 */
+	res =  TEE_SUCCESS;
+
+out:
+	/* TODO: clean-up the memories in case of fail */
+	TEE_Free(ctx->hash_table);
+
+	return res;
+}
+
+static TEE_Result remoteproc_load_fw(uint32_t pt,
+				     TEE_Param params[TEE_NUM_PARAMS])
+{
+	const uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_INPUT,
+						TEE_PARAM_TYPE_MEMREF_INPUT,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_NONE);
+	struct remoteproc_context *ctx = NULL;
+	uint32_t fw_id = params[0].value.a;
+	TEE_Result res = TEE_ERROR_GENERIC;
+
+	if (pt != exp_pt)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	ctx = remoteproc_find_firmware(fw_id);
+	if (!ctx)
+		ctx = remoteproc_add_firmware(fw_id);
+	if (!ctx)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	if (ctx->state != REMOTEPROC_OFF)
+		return TEE_ERROR_BAD_STATE;
+
+	if (!params[1].memref.buffer || !params[1].memref.size)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	DMSG("Got base addr: %p size %x", params[1].memref.buffer,
+	     params[1].memref.size);
+
+	res = remoteproc_verify_firmware(ctx, params[1].memref.buffer,
+					 params[1].memref.size);
+	if (res) {
+		EMSG("Can't Authenticate the firmware, res = %#"PRIx32, res);
+		goto out;
+	}
+
+	res = remoteproc_load_elf(ctx);
+	if (res)
+		goto out;
+
+	ctx->state = REMOTEPROC_LOADED;
+
+out:
+	/* Clear reference to firmware image from shared memory */
+	ctx->fw_img = NULL;
+	ctx->fw_img_sz =  0;
```

Comment:
Replace the 2 space chars with a single one.

There are several other occurrences of 2 space chars that could be removed in this P-R:
ta/remoteproc/remoteproc_core.c lines 112, 324, 424, 562, 576, 583, 743, 976, 1030 and 1031, 
lib/libutee/include/remoteproc_pta.h lines 93 and 171
ta/remoteproc/include/elf_parser.h lines 39 to 42, line 49
core/include/drivers/stm32_remoteproc.h lines 13 and 52